src/SB/Game/zEntDestructObj.cpp

```diff
@@ -28,9 +28,9 @@ namespace
             RwMatrixCopyMacro(modelInst->Mat, ent->model->Mat);
             
             modelInst->Flags &= 0xBBFC;
-            modelInst->Flags |= (s16) (ent->model->Flags & 0x4403);
+            modelInst->Flags |= (u16) (ent->model->Flags & 0x4403);
             ent->model = modelInst;
-            iBoxForModelLocal((xBox*) &ent->bound.box, (xModelInstance *) modelInst);
+            iBoxForModelLocal((xBox*) &ent->bound.box.box, (xModelInstance *) modelInst);
```

Comment:
We can get rid of the cast to `xBox*` now I think.

Reply:
While you're at it, it looks like `modelInst` doesn't need to be casted either.

Reply:
Done